projects/github-actions/repo-gardening/src/utils/get-next-valid-milestone.js

```diff
@@ -61,9 +61,18 @@ async function getNextValidMilestone( octokit, owner, repo, plugin = 'jetpack' )
 			const match = m.description?.match( /(?:Code Freeze|Branch Cut): (\d{4}-\d{2}-\d{2})/ );
 			return ! ( match && moment( match[ 1 ] ) < moment() );
 		} )
-		.sort( ( m1, m2 ) =>
-			compareVersions( m1.title.split( '/' )[ 1 ], m2.title.split( '/' )[ 1 ] )
-		);
+		.sort( ( m1, m2 ) => {
+			const v1 = m1.title.split( '/' )[ 1 ];
+			const v2 = m2.title.split( '/' )[ 1 ];
+
+			// `next` is placeholder version name that represents the immediate next version.
+			// It should be the first item in the list.
+			if ( v1 === 'next' ) return -1;
+			if ( v2 === 'next' ) return 1;
+
+			// Otherwise compare versions normally
+			return compareVersions( v1, v2 );
+		} );
```

Comment:
If I have 2 miletones, one with a version number and a code freeze date, and one `x/next` one with no info, which milestone should we use when the PR gets merged?

Reply:
In our workflow, we wouldn't have multiple open milestones simultaneously and would only have the `boost/next`. 

But @dilirity and I chatted about your question and realized we could modify our workflow slightly to avoid the `boost/next` milestone. Previously, we would rename the milestone to `boost/x.y.z` during code freeze when the version number would be known.

Now, we are going to keep an open milestone for the next minor release instead of `boost/next` as the releases are usually minor. If we need to do a patch/major release, we will rename that milestone. This way, the repo gardening doesn't need to change and we would align more with the rest of Jetpack.